tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11982,6 +12014,33 @@ VERIFY_SUCCEEDED(DoArraysMatch<T>(OutputVector, ExpectedVector,
                                   TestConfig.Tolerance));
 }
 
+#if HAVE_COOPVEC_API
+// Helper class to enable and disable experimental features for cooperative
+// vector tests. Uses RAII to ensure experimental features are enabled/disabled
+// when the object is created/destroyed.
```

Comment:
I have a feeling that this experimental features stuff is probably more engineered that we need.  However, I think this will likely be moot since cooperative vectors probably won't be considered experimental when we take this code to main.